src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -21,66 +21,53 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Tagging;
 /// tracked. That way you can query for intersecting/overlapping spans in a different snapshot
 /// than the one for the tag spans that were added.
 /// </summary>
-internal sealed partial class TagSpanIntervalTree<TTag>(
-    ITextBuffer textBuffer,
-    SpanTrackingMode trackingMode,
-    IEnumerable<TagSpan<TTag>>? values1 = null,
-    IEnumerable<TagSpan<TTag>>? values2 = null) where TTag : ITag
+internal sealed partial class TagSpanIntervalTree<TTag>(SpanTrackingMode spanTrackingMode) where TTag : ITag
 {
-    private readonly ITextBuffer _textBuffer = textBuffer;
-    private readonly SpanTrackingMode _spanTrackingMode = trackingMode;
-    private readonly IntervalTree<TagSpan<TTag>> _tree = IntervalTree.Create(
-        new IntervalIntrospector(textBuffer.CurrentSnapshot, trackingMode),
-        values1, values2);
-
-    private static SnapshotSpan GetTranslatedSpan(
-        TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot, SpanTrackingMode trackingMode)
+    // Tracking mode passed in here doesn't matter (since the tree is empty).
+    public static readonly TagSpanIntervalTree<TTag> Empty = new(SpanTrackingMode.EdgeInclusive);
+
+    private readonly SpanTrackingMode _spanTrackingMode = spanTrackingMode;
+    private readonly IntervalTree<TagSpan<TTag>> _tree = IntervalTree<TagSpan<TTag>>.Empty;
+
+    public TagSpanIntervalTree(
+        ITextSnapshot textSnapshot,
+        SpanTrackingMode trackingMode,
+        IEnumerable<TagSpan<TTag>>? values1 = null,
+        IEnumerable<TagSpan<TTag>>? values2 = null)
+        : this(trackingMode)
     {
-        var localSpan = originalTagSpan.Span;
-
-        return localSpan.Snapshot == textSnapshot
-            ? localSpan
-            : localSpan.TranslateTo(textSnapshot, trackingMode);
+        _tree = IntervalTree.Create(
+            new IntervalIntrospector(textSnapshot, trackingMode),
+            values1, values2);
     }
 
-    private TagSpan<TTag> GetTranslatedITagSpan(TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot)
```

Comment:
removed.  no need for this (the interface version) and GetTranslatedTagSpan now that everything moved to being TagSpan based.